doc/designs/sysaccounts.md

```diff
@@ -0,0 +1,184 @@
+Architect` role by default as well.
+
+### Access controls
+
+To simplify administration of the system accounts, a new privilege `System Account Administrators`
+is added. This privilege is granted by default to the `Security Architect` role.
+
+The system account still needs to be permitted to modify user passwords through a role that includes
+a corresponding permission ('System: Change User password'), through the privilege system. The
+following sequence will allow a system account `my-app` to change user passwords:
+
+```bash
+$ kinit admin
+$ ipa sysaccount-add 'my-app' --random
+$ ipa privilege-add 'my-app password change privilege'
+$ ipa privilege-add-permission 'my-app password change privilege' \
+                  --permission 'System: Change User password'
+$ ipa role-add 'my-app role'
+$ ipa role-add-privilege 'my-app role' \
+             --privilege 'my-app password change privilege'
+$ ipa role-add-member 'my role' --sysaccounts 'my-app'
+```
+
+## Feature Management
+
+### CLI
+
+| Command            | Options | Description                                    |
+| ------------------ | ------- | ---------------------------------------------- |
+| sysaccount-add     | name    | Create LDAP object for a new system account    |
+| sysaccount-del     | name    | Remove LDAP object used for the system account |
+| sysaccount-find    | [name]  | Return list of existing system accounts        |
+| sysaccount-mod     | name    | Modify settings for the system account         |
+| sysaccount-enable  | name    | Allow use of system account for LDAP BIND      |
+| sysaccount-disable | name    | Disable use of system account for LDAP BIND    |
+
+Both the command `sysaccount-add` and `sysaccount-mod` provide an additional
+`--update-without-reset=TRUE|FALSE` option. This option is used to allow a system account to update
+user passwords without forcing them being requested to change on the next login by the user. This is
+done by adding the DN of this system account to the list of DNs allowed to manage password
+synchronization (`passSyncManagersDNs` LDAP attribute) of the `ipa_pwd_extop` plugin.
+
+In order to allow adding system accounts to roles, `role` object has been extended to allow
+membership of system accounts. The command `ipa role-add-member 'my-role' --sysaccounts
+'my-account'` will make sure to grant `my-account` system account all privileges associated with the
+`my-role` role.
+
+Due to the fact that `passSyncManagersDNs` configuration setting is not replicated across IPA
+servers, the `sysaccount-mod` command execution will issue a warning to make sure to update other
+IPA servers as well. System accounts often used by the third-party software to modify passwords; if
+these applications require communication with other IPA servers, an update must be executed against
+those servers.
+
+As IPA command line tool, `ipa`, does not currently provide an option to choose which server to run
```

Comment:
This applies to client context used on an IPA client, but not for an IPA server (client or server context), right?

Reply:
Wouldn't it be better to limit the modification of system accounts to servers, so that it is clear where it was done?

Reply:
A user who executes either `sysaccount-add` or `sysaccount-mod` with `--update-without-reset` option (regardless of the value) will get a message back that tells him about the server the operation was performed on, that the LDAP server on it must be restarted, and that if required, the operations must be performed on other servers as well. This is part of the code already.